File: KotlinCoffeeShop/Kwon/src/main/java/coffeShop/shopdata/MenuInfo.java
package coffeShop.shopdata;

import java.util.HashMap;

public enum MenuInfo {
    esspresso {
        @Override
        public int getPrice() {
            return 200;
        }
    },
    americano {
        @Override
        public int getPrice() {
            return 300;
        }
    },
    capuchino {
        @Override
        public int getPrice() {
            return 400;
        }
    },
    cafelatte {
        @Override
        public int getPrice() {
            return 410;
        }
    };

    public abstract int getPrice();

}

File: KotlinCoffeeShop/Kwon/src/main/kotlin/coffeShop/shopdata/MenuInfo.kt
package coffeShop.shopdata

enum class MenuInfo(val price: Int,val discription: String,val type: MenuType) {

    //Bverage
    ESPRESSO(330,"케냐에서 공수한 아름다운 풍미", MenuType.BEVERAGE),
    AMERICANO(340,"윗메뉴에 물탄맛", MenuType.BEVERAGE),
    CafeLatte(400,"윗메뉴에 우유탄맛", MenuType.BEVERAGE),
    Capuchino(420,"윗메뉴에 뭘탄맛", MenuType.BEVERAGE),

    //Food
    BAGEL(200,"밀가루, 소금맛", MenuType.FOOD),
    WAFFLE(200,"와플맛", MenuType.FOOD),
    CHEESECAKE(400,"치즈맛", MenuType.FOOD),
    SANDWICH(300,"한끼식사", MenuType.FOOD)

}